java-api-core/src/com/kinvey/java/store/requests/data/save/SaveRequest.kt

```diff
@@ -71,12 +71,12 @@ class SaveRequest<T : GenericJson>(private val cache: ICache<T>?, private val ne
                         // to remove the entity from the local cache with the temporary ID.
                         cache?.delete(id)
                     }
+                    ret?.let { cache?.save(it) }
                 } catch (e: IOException) {
                     syncManager?.enqueueRequest(networkManager?.collectionName,
                     networkManager, if (bRealmGeneratedId == true) SyncRequest.HttpVerb.POST else SyncRequest.HttpVerb.PUT, id)
-                    throw e
+                    //throw e
```

Comment:
check with @Martin-Hristov but I believe you also need to check if the http response code is `401`, if so, the error should be throw